packages/drivers/odsp-driver/src/odspDocumentServiceFactory.ts

```diff
@@ -35,3 +40,70 @@ export class OdspDocumentServiceFactory
         );
     }
 }
+
+export class LocalOdspDocumentServiceFactory extends OdspDocumentServiceFactoryCore {
+    constructor(
+        private readonly fluidFile: Uint8Array | string,
+    ) {
+        super(
+            async (_options) => { return null; }, // TODO
+            undefined,
+            async () => getSocketIo(), // TODO
+            undefined,
+            undefined, // TODO
+        );
+    }
+
+    public async createContainer(
+        _createNewSummary: ISummaryTree | undefined,
+        _createNewResolvedUrl: IResolvedUrl,
+        logger?: ITelemetryBaseLogger,
+        _clientIsSummarizer?: boolean,
+    ): Promise<IDocumentService> {
+        const odspLogger = createOdspLogger(logger);
+
+        return this.createDocumentServiceCore(
+            this.getFakeOdspResolvedUrl(),
+            odspLogger,
+        );
+    }
+
+    protected async createDocumentServiceCore(
+        resolvedUrl: IResolvedUrl,
+        odspLogger: TelemetryLogger,
+        _cacheAndTrackerArg?: ICacheAndTracker,
+        _clientIsSummarizer?: boolean,
+    ): Promise<IDocumentService> {
+        return new LocalOdspDocumentService(this.getFakeOdspResolvedUrl(), odspLogger, this.fluidFile);
+    }
+
+    private getFakeOdspResolvedUrl(): IOdspResolvedUrl {
+        return {
+            type: "fluid",
+            odspResolvedUrl: true,
+            id: "1",
+            siteUrl: "fakeUrl",
+            driveId: "1",
+            itemId: "1",
```

Comment:
I'd put some unique numbers here such that if they show up somewhere in telemetry, and someone (accidently) did a search for them in code, he/she would be brought to this code.
Same for URIs below. Given that they all are strings, we can use same unique string everywhere.